library/alloc/src/collections/fixed_queue/mod.rs

```diff
@@ -0,0 +1,288 @@
+use crate::vec::Vec;
+use core::{
+    fmt, mem,
+    ops::{Index, Range},
+    slice,
+};
+
+#[cfg(test)]
+mod tests;
+
+/// Fixed Size Queue:
+/// A linear queue implemented with a static ring buffer of owned nodes.
+///
+/// The `FixedQueue` allows pushing and popping elements in constant time.
+///
+/// The "default" usage of this type is to use [`push`] to add to
+/// the queue, and [`pop`] to remove from the queue. Iterating over
+/// `FixedQueue` goes front to back.
+///
+/// A `FixedQueue` with a known list of items can be initialized from an array:
+/// ```
+/// #![feature(fixed_queue)]
+/// use alloc::collections::FixedQueue;
+///
+/// let fque = FixedQueue::from([1, 2, 3]);
+/// ```
+///
+/// Since `FixedQueue` is an array ring buffer, its elements are contiguous
+/// in memory.
+///
+/// [`push`]: FixedQueue::push
+/// [`pop`]: FixedQueue::pop
+#[derive(Debug)]
+#[unstable(feature = "fixed_queue", issue = "126204")]
+pub struct FixedQueue<T, const N: usize> {
+    buffer: [Option<T>; N],
```

Comment:
This representation is... bad. It requires the **memory leak** in `Index<Range<usize>>` to yield `[T]`. Please come back with a version that uses `MaybeUninit` instead:
```suggestion
    buffer: [MaybeUninit<T>; N],
```

Reply:
This would, of course, require much more `unsafe` code in order to implement correctly, with which an overconfident yes-man cannot help. I strongly advise not even trying to ask ChatGPT, Codex, Copilot, or any other code-generation tool that you may or may not be using about `unsafe` code. They will give you wrong answers, or bad ones.